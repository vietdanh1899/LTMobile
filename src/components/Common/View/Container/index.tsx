import React, { PureComponent } from 'react';
import { StyleSheet } from 'react-native';
import { withTheme, ThemeProps } from 'react-native-elements';
import QuickView, { QuickViewProps } from '../QuickView';

export interface ContainerProps extends QuickViewProps {
  theme?: any;
  secondary?: boolean;
}

class Container extends PureComponent<ContainerProps> {
  render() {
    const {
      backgroundColor: backgroundColorProp,
      secondary,
      theme,
      style,
      children,
      ...otherProps
    } = this.props;

    const backgroundColor = backgroundColorProp
    || (secondary ? theme.colors.bgColorSecondary : theme.colors.bgColor);
    const containerStyle = StyleSheet.flatten([
      { flex: 1, backgroundColor },
      style,
    ]);

    return (
      <QuickView {...otherProps} style={containerStyle}>
        {children}
      </QuickView>
    );
  }
}

export default withTheme(Container as React.ComponentType<ContainerProps & ThemeProps<any>>);
