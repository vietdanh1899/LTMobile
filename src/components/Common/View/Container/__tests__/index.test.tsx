import React from 'react';
import {
  testStyleProps, basicTest, EnumComponent,
} from '@utils/testHelper';
import Container from '..';

const target = <Container />;
const type = EnumComponent.HOC;

describe('Container Component', () => {
  basicTest(target, type);

  it('should apply all style-related props', () => {
    const props = {
      backgroundColor: '#FF8C00',
    };
    testStyleProps(target, props, type, 'style');
  });
});
