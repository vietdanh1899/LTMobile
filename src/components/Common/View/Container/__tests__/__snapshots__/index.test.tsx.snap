// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should render without issues 1`] = `
<QuickView
  column={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  touchableComponent="TouchableOpacity"
/>
`;
